### PR TITLE
[TACHYON-1278] Use replay cache to handle client retries for createRawTable

### PR DESCRIPTION
**Pull request by aaudiber**
https://tachyon.atlassian.net/browse/TACHYON-1278

### CODE REVIEW

File: common/src/main/java/tachyon/retry/CacheUtils.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  public static <T> Cache<String, T> createCache() {
+    return createCache(DEFAULT_MAX_SIZE, DEFAULT_EXPIRE_MS);
+  }
+
+  public static <T> Cache<String, T> createCache(long maxSize, long expireMs) {
+    // TODO(andrew) another constructor which understands CacheBuilder
+    return CacheBuilder.newBuilder().maximumSize(maxSize)
+        .expireAfterWrite(expireMs, TimeUnit.MILLISECONDS).<String, T>build();
+  }
```
Comment (yupeng9) [inline common/src/main/java/tachyon/retry/CacheUtils.java]: Can you add a private constructor?

File: common/src/main/java/tachyon/retry/CacheUtils.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+public final class CacheUtils {
+  // TODO(andrew) Make these configurable
+  private static final long DEFAULT_MAX_SIZE = 10000;
+  private static final long DEFAULT_EXPIRE_MS = 2000;
+
+  public static <T> Cache<String, T> createCache() {
+    return createCache(DEFAULT_MAX_SIZE, DEFAULT_EXPIRE_MS);
+  }
+
+  public static <T> Cache<String, T> createCache(long maxSize, long expireMs) {
```
Comment (yupeng9) [inline common/src/main/java/tachyon/retry/CacheUtils.java]: Can you javadocs?

File: servers/src/main/java/tachyon/master/rawtable/RawTableMasterServiceHandler.java [hunk 3]
```diff
@@ -46,6 +55,21 @@ public long createRawTable(String path, int columns, ByteBuffer metadata)
   }
 
   @Override
+  public long createTable(final String path, final int columns, final ByteBuffer metadata, String nonce)
+      throws TachyonTException, ThriftIOException, TException {
+    try {
+      return ((Long) replayCache.get(nonce, new Callable<Object>() {
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/rawtable/RawTableMasterServiceHandler.java]: Do you think we need to apply this pattern to other non-idempotent operations like mkdir?

File: common/src/main/java/tachyon/retry/CacheUtils.java (hunk 1 above)
Comment (aaudiber) [inline common/src/main/java/tachyon/retry/CacheUtils.java]: Done

File: common/src/main/java/tachyon/retry/CacheUtils.java (hunk 2 above)
Comment (aaudiber) [inline common/src/main/java/tachyon/retry/CacheUtils.java]: Done

File: servers/src/main/java/tachyon/master/rawtable/RawTableMasterServiceHandler.java (hunk 3 above)
Comment (aaudiber) [inline servers/src/main/java/tachyon/master/rawtable/RawTableMasterServiceHandler.java]: Definitely - for now I'm just doing it in this one place to fix the flaky build. When we extend the pattern to all non-idempotent operations we can have more discussion about the best way to do it.

**Comment by aaudiber** [comment]
Moving to https://github.com/amplab/tachyon/pull/2075

**Pull request closed without merging**